### PR TITLE
Issue #601 Adjust expected output in integration tests to match output of setup cmd

### PR DESCRIPTION
**Pull request by jsliacan**
Did what it says on the tin, plus changed `virtualbox` hyperviser to `hyperkit` when passed as `-d` option to `crc start` in basic tests.

### CODE REVIEW

**Comment by jsliacan** [comment]
retest this please

File: test/integration/features/basic.feature [hunk 1]
```diff
@@ -35,34 +35,25 @@ Feature: Basic test
         And stdout should contain "Setting up KVM"
         And stdout should contain "Installing libvirt service and dependencies" 
         And stdout should contain "Adding user to libvirt group"
-        And stdout should contain "Will use root access: add user to libvirt group"
         And stdout should contain "Enabling libvirt"
         And stdout should contain "Starting libvirt service"
-        And stdout should contain "Will use root access: start libvirtd service"
```
Comment (cfergeau) [inline test/integration/features/basic.feature]: I don't understand why these would not be part of stdout?
```
INFO Starting libvirt 'crc' network               
INFO Checking if NetworkManager is installed      
INFO Checking if NetworkManager service is running 
INFO Writing Network Manager config for crc       
INFO Writing dnsmasq config for crc               
INFO Will use root access: write dnsmasq configuration in /etc/NetworkManager/dnsmasq.d/crc.conf 
INFO Will use root access: execute systemctl daemon-reload command 
INFO Will use root access: execute systemctl stop/start command 
```
This is what I get with a `crc setup` on my test machine.

Comment (jsliacan) [reply test/integration/features/basic.feature]: @cfergeau it looks like the output does not contain the `Will use root access` even on CI (see CentosCI console output). So, I guess, it'll be necessary to remove the `/etc/NetworkManager/dnsmasq.d/crc.conf` file before the feature if we want to test it from fresh.

Comment (cfergeau) [reply test/integration/features/basic.feature]: if you look at https://github.com/code-ready/crc/pull/587 , CI was successful and the integration tests have "Will use root access" https://github.com/cfergeau/crc/blob/win-lint/test/integration/features/basic.feature so I don't understand what's going on :-/
Removing `/etc/NetworkManager/dnsmasq.d/crc.conf` just causes one of the tests to show the "Will use root access" message, not all of them :(

Comment (jsliacan) [reply test/integration/features/basic.feature]: > Removing /etc/NetworkManager/dnsmasq.d/crc.conf just causes one of the tests to show the "Will use root access" message, not all of them :(
This would be enough, right? We want to test that line of execution at least once, so exactly once would be good I guess.

I'm a fair bit confused too! Will look around a bit more for some nicer solution.

Comment (cfergeau) [reply test/integration/features/basic.feature]: > This would be enough, right? We want to test that line of execution at least once, so exactly once would be good I guess.

We also want to test `crc setup` on a blank machine, I was thinking centos CI would do that. If you tell me the "Will use root access:" lines need to be removed, that means we are not testing at all the host machine configuration `crc` is supposed to be doing.
If there is one `crc setup` test checking this, and this is about running `crc setup` a second time on the same host, then it's fine.

Comment (jsliacan) [reply test/integration/features/basic.feature]: @cfergeau sorry, I'm confused. I want to remove the file `/etc/NetworkManager/dnsmasq.d/crc.conf` before the testsuite, not the lines containing `Will use root access:` from the test itself. Then inside the tests, the first time `crc setup` is executed, I'll expect `Will use root access:` lines. The second (and later) time `crc setup` is executed in the tests, I will not expect the `Will use root access:` lines. Which part of configuration that `crc` is supposed to be doing wouldn't be tested this way? Cheers!

Comment (cfergeau) [reply test/integration/features/basic.feature]: The very first time `crc setup` is run, you should get `Will use root access:` lines even without removing `/etc/NetworkManager/dnsmasq.d/crc.conf` before. If you need to remove it, you'll likely only get one such line, rather than multiple ones.
As I was saying in https://github.com/code-ready/crc/pull/602#discussion_r326082144 , I don't think this is needed as we have CI currently passing with the integration test checking for "Will use root access:" and not complaining about it.
So I don't understand what changed and what we are discussing exactly :-/

Comment (jsliacan) [reply test/integration/features/basic.feature]: Cool, I just assumed that "The very first time" means literally only once on each machine. Then simulating "the very first time" by removing the file before the test would make sense.

> If you need to remove it, you'll likely only get one such line, rather than multiple ones.

I got all three when I checked this. Hmmm...

> we have CI currently passing with the integration test checking for "Will use root access:" and not complaining about it.

Agreed, that's the puzzle now.

Comment (cfergeau) [reply test/integration/features/basic.feature]: > Cool, I just assumed that "The very first time" means literally only once on each machine. Then simulating "the very first time" by removing the file before the test would make sense.

Just one more clarification on this. Each preflight check/fix is independent, if you remove `/etc/NetworkManager/dnsmasq.d/crc.conf` then the preflight check for this file existence will fail, and the fix will run. Other unrelated checks/fixes won't be run (for example, the codepath doing "sudo systemctl enable libvirtd" won't be exercised).

> 
> > If you need to remove it, you'll likely only get one such line, rather than multiple ones.
> 
> I got all three when I checked this. Hmmm...
> 
> > we have CI currently passing with the integration test checking for "Will use root access:" and not complaining about it.
> 
> Agreed, that's the puzzle now.

File: test/integration/features/#basic.feature# [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    Scenario: CRC version
+        When executing "crc version" succeeds
+        Then stderr should be empty
+        And stdout should contain "version:"
+
+    @darwin @linux @windows
+    Scenario: CRC help
+        When executing "crc --help" succeeds
+        Then stdout should contain "Usage:"
+        And stdout should contain "Available Commands:"
```
Comment (gbraad) [inline test/integration/features/#basic.feature#]: we might need to check for a subset of commands that should NEVER be hidden/always available.

or perhaps even exclude certain things (since with Minishift we accidently added test commands).

File: test/integration/features/basic.feature (hunk 1 above)
Comment (gbraad) [reply test/integration/features/basic.feature]: so machines aren't really clean?

File: test/integration/features/#basic.feature# (hunk 2 above)
Comment (jsliacan) [reply test/integration/features/#basic.feature#]: I added checks for `help` and `version`, as well as `setup`, `start`, `stop`, `delete`, and `status`. Seemed like the "core" ones.

**Comment by jsliacan** [comment]
For completeness: the `Scenario` about `setup` on Windows is being addressed in #543 .

File: test/integration/features/basic.feature (hunk 1 above)
Comment (jsliacan) [reply test/integration/features/basic.feature]: @cfergeau @gbraad I haven't encountered a case where `Will use root access:` wasn't part of the output. So the issue seems to have gone away for the time being. This also makes the changes that this PR introduces much smaller - but there are some.

File: centos_ci.sh [hunk 3]
```diff
@@ -124,14 +124,14 @@ function upload_logs() {
...
-    exit 1
-  fi
+	set +e
+	# In Jenkins slave we have pull secret file in the $HOME/payload/crc_pull_secret
+	# this is copied over using https://github.com/minishift/minishift-ci-jobs/blob/master/minishift-ci-index.yaml#L99
+	make integration BUNDLE_LOCATION=$HOME/Downloads/$BUNDLE PULL_SECRET_FILE=$HOME/payload/crc_pull_secret
+	if [[ $? -ne 0 ]]; then
+		upload_logs $1
+		exit 1
+	fi
```
Comment (praveenkumar) [inline centos_ci.sh]: @jsliacan  This changes shouldn't be part of this PR. it is just matter of test.

File: test/integration/features/basic.feature [hunk 4]
```diff
@@ -53,6 +69,8 @@ Feature: Basic test
         Then stdout should contain "Checking if running as non-root" 
         And stdout should contain "Caching oc binary"
         And stdout should contain "Setting up virtualization"
+        And stdout should contain "Will use root access: change ownership"
+        And stdout should contain "Will use root access: set suid"
         And stdout should contain "Setting file permissions"
```
Comment (cfergeau) [inline test/integration/features/basic.feature]: Up to this point, none of the changes in this commit seem mac-related?

File: test/integration/features/basic.feature [hunk 5]
```diff
@@ -35,16 +42,25 @@ Feature: Basic test
...
         And stdout should contain "Writing Network Manager config for crc"
+        And stdout should contain "Will use root access: write NetworkManager config in /etc/NetworkManager/conf.d/crc-nm-dnsmasq.conf"
+        And stdout should contain "Will use root access: execute systemctl daemon-reload command"
+        And stdout should contain "Will use root access: execute systemctl stop/start command"
         And stdout should contain "Writing dnsmasq config for crc"
+        And stdout should contain "Will use root access: write dnsmasq configuration in /etc/NetworkManager/dnsmasq.d/crc.conf"
+        And stdout should contain "Will use root access: execute systemctl daemon-reload command"
+        And stdout should contain "Will use root access: execute systemctl stop/start command"
+        And stdout should contain "Unpacking bundle from the CRC binary"
         And stdout should contain "Setup is complete"
```
Comment (cfergeau) [inline test/integration/features/basic.feature]: Up to this point, none of the changes in this commit  seem mac-related?

Comment (jsliacan) [reply test/integration/features/basic.feature]: @cfergeau yes, that should be correct. Why is it bad?

Comment (cfergeau) [reply test/integration/features/basic.feature]: They are part of a commit whose log is ` extra: changed virtualbox to hyperkit in crc start for mac with -d...` so it's not really clear why they are in this commit, what they are going to achieve, ...

Comment (jsliacan) [reply test/integration/features/basic.feature]: @cfergeau updated & split into 3 commits topic-wise. Thanks for noticing
